src/main/java/teammates/common/util/Version.java

```diff
@@ -57,18 +59,29 @@ public Version(String versionInString) {
      */
     @Override
     public boolean equals(Object anotherVersion) {
+        if (this == anotherVersion) {
+            return true;
+        }
         if (anotherVersion == null) {
             return false;
         }
-        return toString().equals(anotherVersion.toString());
+        if (this.getClass() != anotherVersion.getClass()) {
+            return false;
+        }
+
+        Version that = (Version) anotherVersion;
+
+        return Objects.equals(this.originalRepresentation, that.originalRepresentation)
```

Comment:
If you look at the previous implementation `toString().equals(anotherVersion.toString());`, you'd notice that only `originalRepresentation` is taken into account here.